### PR TITLE
#87 - login page improvements

### LINKED ISSUE #30: Improvements for register/login section
I was testing register/login page and I have couple of suggestions:

- [x] When user successfully registers, there is no notification about it - I had to guess that I receive email with activation link
- [x] I wanted to reset password, so I click "Nie pamiętam hasła" and... I was redirected to register page
- [x] ~~It should be option on login form like "Zapamiętaj mnie"~~ moved to #89 
- [x] Validation errors should be in Polish, not in English
- [x] When user logins into app, he should see some dashboard or something - now nothing happens
- [x] What will happen, when user doesn't confirm email, but he wants to login into app? Now nothing happens, but maybe he should see some information that he needs to confirm email before redirect to dashboard? Or maybe he should login into app and on dashboard we should display some information about it? We have to think about it
- [x] ~~We should change email design - now they have default style and in English - emails should be also translated~~ moved to #90
- [x] #64
- [x] When I click on field "Szkoła" I see this - it shouldn't be like that

image

- [x] #87
"Akceptuję regulamin i politykę prywatności" and checkbox should be checked or unchecked
- [x] Add page title to browser card
- [x] When clicking on a searchbar, make the icon change the color as well
- [x] Double-clicking Sign In/Up buttons creates an error. Fix it.
- [x] Currently, clicking on the "Login" and "Sign in" buttons in header doesn't get you derictly to the authentication section, if the height of the screen is too small. Fix it 
- [x] When switching from register to login, the page resizes itself. Fix it
- [x] When error occurs the field's border doesn't become red
- [x] In password input we should add some toggle to password visibility

### LINKED ISSUE #87: Homepage: wrap checkbox in label

### LINKED ISSUE #94: Email verification page - Fronted
Add design to the email verfication page

### PR DESCRIPTION
**Pull request by JokeUrSelf**
Should close #94
Should close #87 
Should close #30

> **The colors are messed up on screenshots!**

- Changed style of focused inputs
image
- Searchbar logic is now separated from schools data logic. Searchbar can be used anywhere in the app now.
- Searchbar's icon changes color on focus
image
- Fixed the issue with overflowing autocompletetion background
- Fixed the issue with Email autocompletetion appearing on a searchbar
- Show/hide button for a password input added
image
- Checkbox now activates when clicking on a label
- Information button works again
- Changed the look of "Forgot password" page to match the style on a login page
image

### CODE REVIEW

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Can we add title to this page? Now it's a bit... empty 

image

2. Forgot password page and setting a new password page look different - please unify that

image

image


3. Please add title to this page. Also add show/hide password button

image

**Comment by JokeUrSelf** [comment]
Reset password page looks like this now:
image

It is absolutely impossible to add the title of a page to a header with the current layout system.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
I don't understand why we can just add h1 or div above the form with some text? 

Something like that

image

**Comment by JokeUrSelf** [comment]
> I don't understand why we can just add h1 or div above the form with some text?
> 
> Something like that
> 
> image

Too late, I finally figured it out
image

But your description looks nice, will add right away

**Review by EwelinaSkrzypacz — COMMENTED** [review]
Something bad happen with navigation

image


And why you can't do like I show you on the screen in previous comment?

**Comment by JokeUrSelf** [comment]
fixed header and added description to forgot password
image

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Please fix title (also on page with changing password)

image

2. When I enter password and password confirmation that doesn't match I see this :point_down:  Why label "Nowe hasło" is red? Why is it error "Pole e-mail jest wymagane"? :eyes: 

image

**Comment by JokeUrSelf** [comment]
1 - fixed
2 - 
Small backend correction was made. 
The label should be red, but the border is blue for some reason. Fixed that
image

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Why label "Nowe hasło" is red? Why I see error "Link jest niepoprawny" when I enter password confirmation different form password?

image


2. And to be consistent we should albo add description above inputs, like "Wprowadź nowe hasło z którego będziesz korzystać w aplikacji."

**Comment by EwelinaSkrzypacz** [comment]
Conflicts :(

**Review by EwelinaSkrzypacz — CHANGES_REQUESTED** [review]
1. Why on forgot password page, where user fill input with email there is description like that

image


And on reset page, description is wrong too. Also description should be above the fields not next to them

image

2. Still, when I enter password and password confirmation that differ from each other, I see that label "Nowe hasło" is red <I don't know why> and error "Link jest niepoprawny" <why?>

image

**Comment by JokeUrSelf** [comment]
- UI bugs fixed
- yet another page:
image
- the labels are red for the same reason why clicking on label focuses input. The label and input are related. When input is highlighted as an error, the label is highlighted as well.
- About the error "Link jest niepoprawny". The idea is that if you have no email information in your link this error appears, so either you don't have email information in your link or something is wrong on backend. DominikV1 told me to do one change but since you still see this error, assuming your link is correct, there is something wrong on backend

File: resources/js/Pages/Guest/VerifyEmail.vue [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+<template>
+  <Head>
+    <title>{{ headerRef?.titleRef }}</title>
+    <meta name="Przypomnij hasło" content="Przypomnij hasło">
+  </Head>
+
+  <div class="sm:p-6 sm:pt-12 size-full flex flex-col items-center justify-center sm:h-fit gap-6">
+    <div class="text-md font-medium leading-6 text-gray-900 px-[5vw] text-center flex flex-col gap-1">
+      <p>E-mail weryfikacujny został wysłany do twojego konta.</p>
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Guest/VerifyEmail.vue:28]:
```suggestion
      <p>Dziękujemy za rejestrację! Wysłaliśmy wiadomość z linkiem aktywacyjnym na Twój adres e-mail. Sprawdź skrzynkę odbiorczą, a jeśli nie widzisz wiadomości, zajrzyj do folderu SPAM. Kliknij w link aktywacyjny, aby dokończyć proces rejestracji.</p>
```

File: resources/js/Pages/Guest/VerifyEmail.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    </div>
+    <div class="p-6 gap-6 flex flex-col size-full sm:max-w-lg sm:h-fit">
+      <button class="bg-primary text-white font-bold py-3 px-4 rounded-lg disabled:bg-primary/70" @click="sent">
+        Wyślij ponownie link weryfikacyjny
+      </button>
+    </div>
+  </div>
+</template>
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Guest/VerifyEmail.vue:37]:
```suggestion
```

File: resources/js/Pages/Home.vue [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <script lang="ts" setup>
-import {ref, provide, watch} from 'vue'
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Home.vue:2]:
```suggestion
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
For me, when input is in error state, the label shouldn't be red - please look on other web application - most cases only border is red and validation error.

Still, when I enter password and password confirmation that don't match, I see error with bad email. Please fix this - ask Dominik for help or create issue or something

image

**Comment by JokeUrSelf** [comment]
Not only do I not take the "it's not like on the other pages" statement as an argument, but I can also counter it using the same logic:
[Ryanair](https://www.ryanair.com/)
image
https://onlyfans.com/
image
www.google.com
image

YOU create an issue, as I would need to gather additional requirements from you (like the the text that needs to be shown when the error pops up, as I mentioned in my previous response). Why would we play a broken-phone if you can just write the issue yourself, I don't get it.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
I understand that our opinions may differ on the appearance of validation errors, so it's worth discussing this and establishing a common approach. My guidelines are based on best practices that are used in other applications, but I'd be happy to listen to your perspective and see what solution best fits our project.

As for reporting a problem with an erroneous message when comparing passwords, I can of course create an issue. However, I was depending on you to address the issue yourself, since you've already pointed out that additional requirements need to be added - it's a good exercise for understanding the full process.

I don't care about playing “dumb phone” - the point is to be clear and for you to grow by solving such problems. If you have any doubts, you can always reach out to me, and I will also ask Dominik for additional support if needed.

**Comment by JokeUrSelf** [comment]
I'll try to be more cooperative

File: resources/js/Pages/Guest/ResetPassword.vue [hunk 4]
```diff
@@ -23,14 +32,24 @@ function submit() {
...
     <meta name="Zmiana hasła" content="Zmiana hasła">
   </Head>
-
-  <div class="sm:p-6 sm:pt-12 size-full flex justify-center sm:h-fit">
-    <form class="p-6 gap-4 flex flex-col size-full bg-white shadow border sm:rounded-lg sm:max-w-lg sm:h-fit" @submit.prevent="submit">
-      <CustomInput v-model="form.password" label="Nowe hasło" :error="errors?.password" name="password" type="password" />
-      <CustomInput v-model="form.password_confirmation" label="Powtórz hasło" :error="errors?.password_confirmation" name="password_confirmation" type="password" />
+  <div class="sm:p-6 sm:pt-12 size-full flex flex-col items-center justify-center sm:h-fit gap-6">
+    <div class="text-md font-medium leading-6 text-gray-900 px-[5vw] text-center flex flex-col gap-1">
+      <p>Wprowadź nowe hasło z którego będziesz korzystać w aplikacji.</p>
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Guest/ResetPassword.vue:40]:
```suggestion
      <p>Wprowadź nowe hasło, z którego będziesz korzystać w aplikacji.</p>
```

**Comment by EwelinaSkrzypacz** [comment]
And if this pr closes the issue, pleas add this information to pr description

**Review by EwelinaSkrzypacz — COMMENTED** [review]
When I click "Nie pamiętam hasła" I see error

image

**Comment by JokeUrSelf** [comment]
> When I click "Nie pamiętam hasła" I see error
> 
Fixed

**Review by EwelinaSkrzypacz — DISMISSED** [review]

**Comment by EwelinaSkrzypacz** [comment]
Conflicts

**Comment by JokeUrSelf** [comment]
@dawidrudnik please check the PR

File: resources/js/Pages/Guest/ForgotPassword.vue [hunk 5]
```diff
@@ -2,25 +2,33 @@
...
 function submit() {
-  form.post('/auth/forgot-password')}
+  form.post('/auth/forgot-password')
+}
 </script>
 
 <template>
   <Head>
-    <title>Przypomnij hasło</title>
+    <title>{{ headerRef?.titleRef }}</title>
```
Comment (dawidrudnik) [inline resources/js/Pages/Guest/ForgotPassword.vue:22]: Why ? You can just `<Head title="Home">`. I think meta tags is not necessary here. But if you want to keep meta tags, change please title set to some more readable.

File: resources/js/Pages/Guest/ResetPassword.vue [hunk 6]
```diff
@@ -23,14 +32,24 @@ function submit() {
 
 <template>
   <Head>
```
Comment (dawidrudnik) [inline resources/js/Pages/Guest/ResetPassword.vue:34]: Maybe create some wrapper component for that.

File: resources/js/Pages/Guest/VerifyEmail.vue [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<script setup lang="ts">
+import { Head, useForm } from '@inertiajs/vue3'
+import { inject, onMounted, type Ref } from 'vue'
+import type Header from '@/components/Common/Header.vue'
+
+const form = useForm({})
```
Comment (dawidrudnik) [inline resources/js/Pages/Guest/VerifyEmail.vue:6]: Did you need a useForm here if you don't set any values inside. You can use just a router from inertia. 
https://inertiajs.com/manual-visits

File: resources/js/Types/Option.d.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export interface Option{
```
Comment (dawidrudnik) [inline resources/js/Types/Option.d.ts:1]: Do you need export if this is a `d.ts` file ?

File: resources/js/components/Common/Header.vue [hunk 9]
```diff
@@ -3,35 +3,38 @@ import {type Page} from '@/Types/Page'
...
-        <a
-          v-for="item in pages"
-          :key="item.title"
-          :href="item.href"
-          class="hidden md:block font-semibold leading-6 text-gray-900 transition-colors duration-200 hover:text-primary-800"
-          :class="{'text-primary': isSelected(item) }"
-        >
-          {{ item.title }}
-        </a>
+        <span v-if="pages.length <= 0" class="font-semibold leading-6 text-gray-900">{{ titleRef }}</span>
```
Comment (dawidrudnik) [inline resources/js/components/Common/Header.vue:26]: Why `titleRef` is a `ref` ? Maybe just here create some `slot` ? Or `prop` ?

File: resources/js/components/Common/PasswordInput.vue [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+import CustomInput from './CustomInput.vue'
```
Comment (dawidrudnik) [inline resources/js/components/Common/PasswordInput.vue:3]: Imports with `@`

File: resources/js/components/Home/RegisterForm.vue [hunk 11]
```diff
@@ -1,123 +1,73 @@
...
-        <input id="password" v-model="form.password" required name="password" type="password" autocomplete="current-password"
-               :class="{'ring-red focus:ring-red':errors.password}" class="duration-200 ring-inset outline-none focus:ring focus:ring-primary bg-white/30 rounded-lg w-full p-3 text-gray-900 ring-2 ring-primary/30 placeholder:text-gray-400"
-        >
-        <div v-if="errors.password" class="text-red">{{ errors.password }}</div>
-      </div>
-    </div>
+    <PasswordInput v-model="form.password" :error="errors.password" />
 
-    <div class="mx-2 mt-4 flex flex-row items-center gap-4">
+    <label class="mx-2 mt-4 flex flex-row items-center gap-4">
```
Comment (dawidrudnik) [inline resources/js/components/Home/RegisterForm.vue:52]: Label ? But where is `for` defended ?

File: resources/js/Pages/Guest/ForgotPassword.vue (hunk 5 above)
Comment (JokeUrSelf) [reply resources/js/Pages/Guest/ForgotPassword.vue]: Fixed

File: resources/js/Pages/Guest/ResetPassword.vue (hunk 6 above)
Comment (JokeUrSelf) [reply resources/js/Pages/Guest/ResetPassword.vue]: I decided to get rid of meta tags in Head, so no need in extra component

File: resources/js/Pages/Guest/VerifyEmail.vue (hunk 7 above)
Comment (JokeUrSelf) [reply resources/js/Pages/Guest/VerifyEmail.vue:5]: I'm using `form.processing` in this file, it is only available in `useForm`

File: resources/js/Types/Option.d.ts (hunk 8 above)
Comment (JokeUrSelf) [reply resources/js/Types/Option.d.ts:1]: Yes, it doesn't work otherwise

File: resources/js/components/Common/Header.vue (hunk 9 above)
Comment (JokeUrSelf) [reply resources/js/components/Common/Header.vue]: Fixed, changed to `prop`

File: resources/js/components/Home/RegisterForm.vue (hunk 11 above)
Comment (JokeUrSelf) [reply resources/js/components/Home/RegisterForm.vue:52]: You don't need `for` if `label` wraps your input

File: resources/js/Pages/Guest/VerifyEmail.vue (hunk 7 above)
Comment (dawidrudnik) [reply resources/js/Pages/Guest/VerifyEmail.vue:5]: You can also use some event callbacks to handle 'processing' -> https://inertiajs.com/manual-visits.

File: resources/js/Layouts/IconLayout.vue [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<template>
```
Comment (dawidrudnik) [inline resources/js/Layouts/IconLayout.vue:1]: This is no necessary component 
```suggestion
```

File: resources/js/Types/Option.d.ts (hunk 8 above)
Comment (dawidrudnik) [reply resources/js/Types/Option.d.ts:1]: Please me explain why this necessary. Give me example where this is not work and how you use this interface.

Comment (JokeUrSelf) [reply resources/js/Types/Option.d.ts]: Fixed.

Sorry, it works. Global type declarations don't get recognized when passed directly to a macro as type arguments (`defineProps<Opition>`), but here I use `Opition` as inline object type instead, so it works.

Comment (dawidrudnik) [reply resources/js/Types/Option.d.ts]: :rocket:

**Review by dawidrudnik — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**